tests/e2e/kcs/test_monitor_recovery.py

```diff
@@ -0,0 +1,945 @@
+        mon_recovery.monitor_rebuild(mon_map_cmd)
+
+        logger.info("Reverting mon, osd and mgr deployments")
+        mon_recovery.revert_patches(dep_revert)
+
+        logger.info("Scaling back rook and ocs operators")
+        mon_recovery.scale_rook_ocs_operators(replica=1)
+
+        logger.info("Recovering CephFS")
+        mon_recovery.scale_rook_ocs_operators(replica=0)
+        logger.info(
+            "Patching MDSs to remove LivenessProbe and setting sleep to infinity"
+        )
+        mon_recovery.patch_sleep_on_mds()
+        logger.info("Resetting the fs")
+        ceph_fs_recovery()
+        logger.info("Reverting MDS deployments")
+        mon_recovery.revert_patches(mds_revert)
+        logger.info("Scaling back rook and ocs operators")
+        mon_recovery.scale_rook_ocs_operators(replica=1)
+        logger.info("Recovering mcg by re-spinning the pods")
+        recover_mcg()
+        remove_global_id_reclaim()
+        for pod_obj in self.dc_pods:
+            pod_obj.delete(force=True)
+        new_md5_sum = []
+        logger.info("Verifying md5sum of files after recovery")
+        for pod_obj in get_spun_dc_pods(self.dc_pods):
+            pod_obj.ocp.wait_for_resource(
+                condition=constants.STATUS_RUNNING,
+                resource_name=pod_obj.name,
+                timeout=600,
+                sleep=10,
+            )
+            new_md5_sum.append(pod.cal_md5sum(pod_obj, self.filename))
+        logger.info(f"Md5sum calculated after recovery: {new_md5_sum}")
+        if collections.Counter(new_md5_sum) == collections.Counter(self.md5sum):
+            logger.info(
+                f"Verified: md5sum of {self.filename} on pods matches with the original md5sum"
+            )
+        else:
+            assert False, f"Data corruption found {new_md5_sum} and {self.md5sum}"
+        logger.info("Getting object after recovery")
+        assert bucket_utils.s3_get_object(
+            s3_obj=mcg_obj,
+            bucketname=self.bucket_name,
+            object_key=self.object_key,
+        ), "Failed: GetObject"
+
+        # New pvc, dc pods, obcs
+        new_dc_pods = [
+            dc_pod_factory(
+                interface=constants.CEPHBLOCKPOOL,
+            ),
+            dc_pod_factory(
+                interface=constants.CEPHFILESYSTEM,
+            ),
+        ]
+        for pod_obj in new_dc_pods:
+            pod_obj.exec_cmd_on_pod(command=self.dd_cmd)
+        logger.info("Creating new bucket and write object")
+        new_bucket = bucket_factory(interface="OC")[0].name
+        assert bucket_utils.s3_put_object(
+            s3_obj=mcg_obj,
+            bucketname=new_bucket,
+            object_key=self.object_key,
+            data=self.object_data,
+        ), "Failed: PutObject"
+        logger.info("Archiving the ceph crash warnings")
+        tool_pod = get_ceph_tools_pod()
+        tool_pod.exec_ceph_cmd(ceph_cmd="ceph crash archive-all", format=None)
+
```

Comment:
Should we also do some sanity and pod health checks at the end of the recovery to make sure that everything is fine?

Reply:
Ack added